sphinx-mode.el

```diff
@@ -122,10 +123,49 @@ If BUFFER is not given use the `current-buffer'."
     (find-file (plist-get target :file))
     (goto-char (plist-get target :point))))
 
+(defun sphinx-compile-find-makefile-dir () ; adapted from rst-compile-find-conf
+  "Look for the Sphinx Makefile in parents of the current path and returns the directory which contains it as a string."
+  (interactive)
+  (let ((file-name "Makefile")
+        (buffer-file (buffer-file-name)))
+    ;; Move up in the dir hierarchy till we find Makefile.
+    (let* ((dir (file-name-directory buffer-file))
+	   (prevdir nil))
+      (while (and (or (not (string= dir prevdir))
+		      (setq dir nil)
+		      nil)
+                  (not (file-exists-p (concat dir file-name))))
+        ;; Move up to the parent dir and try again.
+	(setq prevdir dir)
+        (setq dir (expand-file-name (file-name-directory
+                                     (directory-file-name
+				      (file-name-directory dir))))))
+      (or (and dir (concat dir "")) nil))))
+
+;; TODO: replace this with sphinx-compile and sphinx-compile-and-view
+(defun sphinx-view-build-html ()
+  "Runs 'make html' in project root directory and then views compiled HTML of current source file."
+  (interactive)
+    (let
+      (
+       (buffer-name-base (file-name-base (buffer-name))) ; reads name of current bufer (presumably a .rst file), minus extension, into buffer-name-base
```

Comment:
the comments probably aren't necessary.

Why are you using `buffer-name` instead of `buffer-file-name`?  People can rename buffers or use something like `uniquify` which would break this.